src/WebJobs.Script.WebHost/Diagnostics/WebHostLoggerProviderFactory.cs

```diff
@@ -20,11 +20,11 @@ public WebHostLoggerProviderFactory(IEventGenerator eventGenerator, ScriptSettin
             _settingsManager = settingsManager;
         }
 
-        public override IEnumerable<ILoggerProvider> CreateLoggerProviders(ScriptHostConfiguration scriptConfig, ScriptSettingsManager settingsManager, Func<bool> isFileLoggingEnabled, Func<bool> isPrimary)
+        public override IEnumerable<ILoggerProvider> CreateLoggerProviders(string hostInstanceId, ScriptHostConfiguration scriptConfig, ScriptSettingsManager settingsManager, Func<bool> isFileLoggingEnabled, Func<bool> isPrimary)
```

Comment:
Can you use the `InstanceId` from the `settingsManager` instead of adding a new param?

Reply:
InstanceId on settingsManager is a static value read from environment variables on the website. We need an Id to keep track of ScripHost instances.

Reply:
here we go again  :)